tools/benchmark-query-engine/main.go

```diff
@@ -339,11 +340,22 @@ func (a *app) runTestCase(name string, printBenchmarkHeader bool) error {
 			}
 		} else if isBenchmarkLine {
 			fmt.Print(l)
-			fmt.Printf("     %v B\n", usage.Maxrss)
+			fmt.Printf("     %v B\n", maxRSSInBytes(usage))
 		} else if !isPassLine {
 			fmt.Println(l)
 		}
 	}
 
 	return nil
 }
+
+func maxRSSInBytes(usage *syscall.Rusage) int64 {
+	switch runtime.GOOS {
+	case "Linux":
+		return usage.Maxrss * 1024 // Maxrss is returned in kilobytes on Linux.
```

Comment:
I'm confused by this comment.. If it's returned in kilobytes, why multiply it by 1024? Wouldn't you divide it?

Reply:
`usage.Maxrss` is given in kilobytes, but we want to return a value in bytes from `maxRSSInBytes`.

For example, if `usage.Maxrss` is 1, then that represents 1 KB, or 1024 bytes.

Reply:
Right, that makes complete sense on re-reading, I'm not sure what I was thinking. I'll blame covid-brain. Sorry for the hold up!

Reply:
No worries 🙂